admin/index.php

```diff
@@ -537,6 +537,23 @@
 	$query = $db->simple_select("adminoptions", "*", "uid='".$mybb->user['uid']."'");
 	$admin_options = $db->fetch_array($query);
 
+	// Only update language / theme once fully authenticated
+	if(empty($admin_options['authsecret']) || $admin_session['authenticated'] == 1)
+	{
+		if(!empty($admin_options['cplanguage']) && file_exists(MYBB_ROOT."inc/languages/".$admin_options['cplanguage']."/admin/home_dashboard.lang.php"))
+		{
+            $cp_language = $admin_options['cplanguage'];
+            $lang->set_language($cp_language, "admin");
+            $lang->load("global"); // Reload global language vars
+            $lang->load("messages", true);
+        }
+
+		if(!empty($admin_options['cpstyle']) && file_exists(MYBB_ADMIN_DIR."/styles/{$admin_options['cpstyle']}/main.css"))
+		{
+            $cp_style = $admin_options['cpstyle'];
+        }
+    }
```

Comment:
Nitpick: Nitpick: indentation.

```suggestion
			$cp_style = $admin_options['cpstyle'];
		}
	}
```